### ISSUE TITLE
cannot remove compiler from imported component

### ISSUE THREAD

**Issue by KutnerBitsrc** [teambit/bit#1721]
I am trying to add the new TS compiler for a component I am using.
The component used to be a .js file and uses the babel@0.0.20 compiler

I am using this command to install the compiler:
```bit import -c bit.envs/compilers/typescript```

## Expected Behavior
component compiler should be:
bit.envs/compilers/typescript@3.0.3

## Actual Behavior
❯ bit show constants/layouts
```
Id			│ constants/layouts@0.0.3
Compiler		│ bit.envs/compilers/babel@0.0.20
Language		│ javascript
Main File		│ layouts.ts
Dependencies		│
Dev Dependencies	│
Files			│ layouts.ts
```

## Steps to Reproduce the Problem
  1. bit init
  1.  bit import bit.rendering/constants/layouts
  1.  bit s
  1.  bit import -c bit.envs/compilers/typescript
  1.  bit build

## Specifications

  - Bit version: 14.1.3
  - Node version: 12.1.0
  - npm / yarn version: node 6.9.0
  - Platform: macOS

**Comment by itaymendel** [comment]
The `bit import -c` command does not update the compiler for imported components.
This is something that needs to be resolved by Bit, but until then - you should manually update the components configuration with the new compiler.

To do so, after you import the component open its `package.json` file, and edit the `compiler` key in the `bit` config object.

**Comment by GiladShoham** [comment]
@KutnerUri If it would be just updated by importing a compiler, it will make a lot of issues, since you might have different components with different compilers in the same workspace, and you don't want just to override all of them.

**Comment by KutnerBitsrc** [comment]
I see.
thanks for the explaination

**Comment by KutnerBitsrc** [comment]
Still happening, and now changing the package.json does not work anymore.
bit just uses the old compiler.

The ```bit import -c xxx``` syntax is confusing and only applies to the whole project.
I would have expected a syntax like this:
```bit attach bit.envs/compilers/typescript --to constants/layouts [--as compiler]```
```bit attach bit.envs/compilers/typescript [--to workspace] [--as compiler]```

**Comment by GiladShoham** [comment]
@KutnerBitsrc We already have an issue about split it to a different command.
You can see it here:
https://github.com/teambit/bit/issues/1898
Feel free to write feedback there:)

**Comment by KutnerBitsrc** [comment]
OH NICE!!

**Comment by GiladShoham** [comment]
@davidfirst I just tried to remove a compiler from an imported component. tried the follow methods:
* change the env.compiler to "none" in component package.json 
* change the env.compiler to "-" in component package.json (got an error that it's not a valid bit id
* add overrides.env.compiler = "-" in component package.json (combined with none in the env.compiler, old value in env.compiler, removed env.compiler)
* add overrides.compName.env.compiler = ="-" in the workspace package.json

none worked.
When I change the env.compiler to a different version, for example, it does work, but change to "none" not working.

**Comment by davidfirst** [comment]
Interesting.
For the workspace config, the minus works. I just verified it.
For the component config, I remembered that removing the compiler object is what needed. However, I just tested it and it doesn't work.

The question is what should work? 'none', minus sign or removing the compiler object.
I wouldn't go with 'none', it's an old notation that no one knows/work with. I vote for minus or removing the object. @itaymendel , @GiladShoham , let me know your opinions.
